### PR TITLE
Improve media queries restrictions and enhance usage developer comfort

### PR DESCRIPTION
**Pull request by kosorz**
I would like to suggest using the following approach with media queries.

Using this approach helps to preserve consistency across the codebase since it's necessary to introduce some minor changes to the media setup to start using a new breakpoint additionally all breakpoints are 'min' ones therefore app styling integrity is less prone to human error.

Also, it's generally easier to style using such queries since it's more concise and requires less effort to achieve the same goal.
image

### CODE REVIEW

**Comment by kosorz** [comment]
I've just spotted an emotion specific problem: 
image
I'll try to resolve it soon!

**Review by kdembler — CHANGES_REQUESTED** [review]
Hey, thanks for this contribution!

We've discussed this approach internally with the team and we found it problematic to work with. The reason for that is the tooling will not really understand what's going on with this custom `media` function. We tested both VSCode and WebStorm and they both don't understand that CSS is supposed to go inside it and they will not offer any code assistance. Same with Prettier - it will not properly parse and format the CSS enclosed in `media`.

For this reason, we will not be accepting the PR in its current form. However, I believe your general idea of reducing possibility of a human error is a good one. I suggest a different, similar approach. We could have pre-defined string for full media query condition. For example, `media.small` would be `"@media screen and (min-width: ${breakpoints.small})"` and would be used like this:
```js
${media.small} {
  color: red;
}
```

Let me know what you think about that

---

One general point I also want to make is that we're happy you're interested in helping out with the codebase and we will gladly accept any contributions. However, what I'd suggest going forward - if you want to make an app-wide change like this, it will be a good idea to first create an issue describing the problem and the solution you have in mind. This way, we all have a chance to discuss things before getting to work. This will surely reduce a chance of any work going to waste.

Thanks

**Comment by kosorz** [comment]
Have you seen [this](https://spectrum.chat/emotion/general/functions-that-are-interpolated-tagged-template-literal-question~7818175a-e88c-4308-a51a-19547fa09523)?
If you would google the error I was receiving you would end up at this spectrum chat.
This approach will allow the tooling to process the code and resolve the errors thrown.

- - -

Actually, I will have to create at least one issue then, it might make some impact on the codebase. I wonder if you will agree with me, stay tuned!

File: src/components/Dialogs/BaseDialog/BaseDialog.style.ts [hunk 1]
```diff
@@ -18,7 +15,11 @@ export const StyledContainer = styled.div`
   padding: var(--dialog-padding);
   box-shadow: 0 8px 8px rgba(0, 0, 0, 0.12), 0 24px 40px rgba(0, 0, 0, 0.16);
 
-  @media screen and (min-height: 800px) {
+  ${media.small} {
+    --dialog-padding: ${sizes(6)};
+  }
```
Comment (kdembler) [inline src/components/Dialogs/BaseDialog/BaseDialog.style.ts]: Let's move this back up, next to the other var declaration

File: src/components/Sidenav/SidenavBase.style.ts [hunk 2]
```diff
@@ -92,9 +93,10 @@ export const SidebarNavItem = styled.li<ExpandableElementProps>`
   &[data-badge]:after {
     left: ${sizes(10)};
     top: ${sizes(3)};
-    @media screen and (min-width: ${breakpoints.medium}) {
-      transform: translateY(${({ expanded }) => (expanded ? 0 : -8)}px);
-      transition: transform ${transitions.timings.regular} ${transitions.easing};
+
+    ${media.medium} {
+      transform: translateY(${({ expanded }: { expanded?: boolean }) => (expanded ? 0 : -8)}px)
+      transition: transform ${transitions.timings.regular} ${transitions.easing}
```
Comment (kdembler) [inline src/components/Sidenav/SidenavBase.style.ts]: Missing semicolons

File: src/components/Sidenav/SidenavBase.style.ts [hunk 3]
```diff
@@ -92,9 +93,10 @@ export const SidebarNavItem = styled.li<ExpandableElementProps>`
   &[data-badge]:after {
     left: ${sizes(10)};
     top: ${sizes(3)};
-    @media screen and (min-width: ${breakpoints.medium}) {
-      transform: translateY(${({ expanded }) => (expanded ? 0 : -8)}px);
-      transition: transform ${transitions.timings.regular} ${transitions.easing};
+
+    ${media.medium} {
+      transform: translateY(${({ expanded }: { expanded?: boolean }) => (expanded ? 0 : -8)}px)
```
Comment (kdembler) [inline src/components/Sidenav/SidenavBase.style.ts]: Types of props are already provided in the component definition `styled.li<ExpandableElementProps>`. We don't need to provide them explicitly here

File: src/components/Sidenav/SidenavBase.style.ts [hunk 4]
```diff
@@ -116,8 +118,8 @@ export const SidebarNavLink = styled(Link, { shouldForwardProp: isPropValid })<S
     background-color: ${({ isStudio }) => (isStudio ? colors.transparentWhite[32] : colors.transparentBlack[54])};
   }
   > svg {
-    @media screen and (min-width: ${breakpoints.medium}) {
-      transform: translateY(${({ expanded }) => (expanded ? 0 : -8)}px);
+    ${media.medium} {
+      transform: translateY(${({ expanded }: { expanded?: boolean }) => (expanded ? 0 : -8)}px);
```
Comment (kdembler) [inline src/components/Sidenav/SidenavBase.style.ts]: Unnecessary prop type

File: src/components/Sidenav/SidenavBase.style.ts [hunk 5]
```diff
@@ -132,13 +134,13 @@ export const SidebarNavLink = styled(Link, { shouldForwardProp: isPropValid })<S
     background-color: ${({ isStudio }) => (isStudio ? colors.transparentWhite[6] : colors.transparentBlack[24])};
   }
   :after {
-    @media screen and (min-width: ${breakpoints.medium}) {
-      content: ${({ content }) => `'${content}'`};
+    ${media.medium} {
+      content: ${({ content }: { content: string }) => `'${content}'`};
```
Comment (kdembler) [inline src/components/Sidenav/SidenavBase.style.ts]: Unnecessary prop type

File: src/components/Sidenav/SidenavBase.style.ts [hunk 6]
```diff
@@ -132,13 +134,13 @@ export const SidebarNavLink = styled(Link, { shouldForwardProp: isPropValid })<S
...
       position: absolute;
       font-size: 12px;
       color: white;
       transition: opacity ${transitions.timings.regular} ${transitions.easing};
-      opacity: ${({ expanded }) => (expanded ? 0 : 1)};
+      opacity: ${({ expanded }: { expanded?: boolean }) => (expanded ? 0 : 1)};
```
Comment (kdembler) [inline src/components/Sidenav/SidenavBase.style.ts]: Unnecessary prop type

File: src/components/Topbar/TopbarBase.style.tsx [hunk 7]
```diff
@@ -55,37 +55,39 @@ export const LogoLink = styled(Link)`
...
 `
 
 export const FullLogo = styled(UnstyledFullLogoDefault)<LogoContainerProps>`
   display: none;
   height: ${sizes(8)};
-  @media screen and (min-width: ${breakpoints.small}) {
-    display: ${({ variant }) => (variant === 'default' ? 'block' : 'flex')};
+
+  ${media.small} {
+    display: ${({ variant }: { variant?: string }) => (variant === 'default' ? 'block' : 'flex')};
```
Comment (kdembler) [inline src/components/Topbar/TopbarBase.style.tsx]: Unnecessary prop type

File: src/shared/components/ActionBar/ActionBar.style.ts [hunk 8]
```diff
@@ -15,13 +15,14 @@ export const StyledActionBarContainer = styled.div<ActionBarContainerProps>`
...
-  @media screen and (min-width: ${breakpoints.medium}) {
+
+  ${media.medium} {
     flex-direction: row;
     justify-content: space-between;
     padding: ${sizes(4)} ${sizes(8)};
   }
 
-  transform: translateY(${({ isActive }) => (isActive ? '0' : '100%')});
+  transform: translateY(${({ isActive }: { isActive?: boolean }) => (isActive ? '0' : '100%')});
```
Comment (kdembler) [inline src/shared/components/ActionBar/ActionBar.style.ts]: Unnecessary prop type

File: src/shared/components/ChannelCover/ChannelCover.style.ts [hunk 9]
```diff
@@ -46,50 +46,55 @@ export const CoverImage = styled.div<CoverImageProps>`
...
+
+  ${media.small} {
     background-image: linear-gradient(
         0deg,
         black 0%,
         black ${Math.min(CONTENT_OVERLAP_MAP.SMALL - GRADIENT_OVERLAP, 0)}px,
         rgba(0, 0, 0, 0) ${CONTENT_OVERLAP_MAP.SMALL - GRADIENT_OVERLAP + GRADIENT_HEIGHT}px
       ),
-      url(${({ src }) => src});
+      url(${({ src }: { src: string }) => src});
```
Comment (kdembler) [inline src/shared/components/ChannelCover/ChannelCover.style.ts]: Unnecessary prop type

File: src/shared/components/ChannelCover/ChannelCover.style.ts [hunk 10]
```diff
@@ -46,50 +46,55 @@ export const CoverImage = styled.div<CoverImageProps>`
...
+
+  ${media.medium} {
     background-image: linear-gradient(
         0deg,
         black 0%,
         black ${Math.min(CONTENT_OVERLAP_MAP.MEDIUM - GRADIENT_OVERLAP, 0)}px,
         rgba(0, 0, 0, 0) ${CONTENT_OVERLAP_MAP.MEDIUM - GRADIENT_OVERLAP + GRADIENT_HEIGHT}px
       ),
-      url(${({ src }) => src});
+      url(${({ src }: { src: string }) => src});
```
Comment (kdembler) [inline src/shared/components/ChannelCover/ChannelCover.style.ts]: Unnecessary prop type

File: src/shared/components/ChannelCover/ChannelCover.style.ts [hunk 11]
```diff
@@ -46,50 +46,55 @@ export const CoverImage = styled.div<CoverImageProps>`
...
+
+  ${media.large} {
     background-image: linear-gradient(
         0deg,
         black 0%,
         black ${CONTENT_OVERLAP_MAP.LARGE - GRADIENT_OVERLAP}px,
         rgba(0, 0, 0, 0) ${CONTENT_OVERLAP_MAP.LARGE - GRADIENT_OVERLAP + GRADIENT_HEIGHT}px
       ),
-      url(${({ src }) => src});
+      url(${({ src }: { src: string }) => src});
```
Comment (kdembler) [inline src/shared/components/ChannelCover/ChannelCover.style.ts]: Unnecessary prop type

File: src/shared/components/ChannelCover/ChannelCover.style.ts [hunk 12]
```diff
@@ -46,50 +46,55 @@ export const CoverImage = styled.div<CoverImageProps>`
...
+
+  ${media.xlarge} {
     background-image: linear-gradient(
         0deg,
         black 0%,
         black ${CONTENT_OVERLAP_MAP.XLARGE - GRADIENT_OVERLAP}px,
         rgba(0, 0, 0, 0) ${CONTENT_OVERLAP_MAP.XLARGE - GRADIENT_OVERLAP + GRADIENT_HEIGHT}px
       ),
-      url(${({ src }) => src});
+      url(${({ src }: { src: string }) => src});
```
Comment (kdembler) [inline src/shared/components/ChannelCover/ChannelCover.style.ts]: Unnecessary prop type

File: src/shared/components/ChannelCover/ChannelCover.style.ts [hunk 13]
```diff
@@ -46,50 +46,55 @@ export const CoverImage = styled.div<CoverImageProps>`
...
+
+  ${media.xxlarge} {
     background-image: linear-gradient(
         0deg,
         black 0%,
         black ${CONTENT_OVERLAP_MAP.XXLARGE - GRADIENT_OVERLAP}px,
         rgba(0, 0, 0, 0) ${CONTENT_OVERLAP_MAP.XXLARGE - GRADIENT_OVERLAP + GRADIENT_HEIGHT}px
       ),
-      url(${({ src }) => src});
+      url(${({ src }: { src: string }) => src});
```
Comment (kdembler) [inline src/shared/components/ChannelCover/ChannelCover.style.ts]: Unnecessary prop type

File: src/shared/components/HeaderTextField/HeaderTextField.style.tsx [hunk 14]
```diff
@@ -44,6 +38,14 @@ export const StyledInput = styled.input<StyledInputProps>`
...
     filter: brightness(80%);
   }
+
+  ${media.small} {
+    --input-max-width: 400px;
+  }
+
+  ${media.medium} {
+    --input-max-width: 600px;
+  }
```
Comment (kdembler) [inline src/shared/components/HeaderTextField/HeaderTextField.style.tsx]: Let's move those back up

File: src/shared/components/VideoPlayer/VideoPlayer.tsx [hunk 15]
```diff
@@ -53,19 +53,17 @@ const VideoPlayer: React.ForwardRefRenderFunction<HTMLVideoElement, VideoPlayerP
       return
     }
 
-    if (playing != null) {
```
Comment (kdembler) [inline src/shared/components/VideoPlayer/VideoPlayer.tsx]: Changes here are out of scope of this PR, feel free to open a new one with some rationale

File: src/components/Dialogs/BaseDialog/BaseDialog.style.ts [hunk 16]
```diff
@@ -18,7 +15,11 @@ export const StyledContainer = styled.div`
...
+
+  ${media.smedium} {
```
Comment (kdembler) [inline src/components/Dialogs/BaseDialog/BaseDialog.style.ts]: Let's just use regular `medium` here and drop the `smedium` breakpoint

**Review by kdembler — CHANGES_REQUESTED** [review]
Looks mostly good, found some unnecessary changes though

**Review by kdembler — APPROVED** [review]
Looks good, thanks!

Will need a rebase before merging though

**Comment by kosorz** [comment]
I think it's ready to be merged.
image

No more actions will be made on my end towards merging this PR.

---

Have a nice day! :)

**Comment by kdembler** [comment]
Merged, thanks!

**Pull request merged**